src/display/editor/tools.js

```diff
@@ -1007,9 +1037,15 @@ class AnnotationEditorUIManager {
       hasSelectedText: true,
     });
 
-    if (this.#mode !== AnnotationEditorType.HIGHLIGHT) {
+    const mustHighlight = this.#mode === AnnotationEditorType.HIGHLIGHT;
+    if (!mustHighlight && this.#mode !== AnnotationEditorType.NONE) {
       return;
     }
+
+    if (!mustHighlight) {
+      this.#highlightToolbar?.hide();
+    }
+
```

Comment:
This isn't all that easy to understand at a glance, so I might be misunderstanding it, however:
Won't this lead to a bunch more code being run, further below, in the `AnnotationEditorType.NONE` case and *if so* is that expected now?